### PR TITLE
Initial metrics

### PR DESCRIPTION
**Pull request by kciesielski**
This PR introduces dropwizard metrics library for collecting test stats.
1. Histogram is used to observe throughput. We increase histogram "bars" for each message in a particular second.
2. Timer is used to observe latency. We calculate time spent on single "request".
3. Data is stored in DynamoDB.
4. One missing part is formatting in `ShowStats`. Once we agree that we have the correct data stored, we can polish this in a separate PR.

### CODE REVIEW

File: src/main/scala/com/softwaremill/mqperf/Receiver.scala [hunk 1]
```diff
@@ -31,36 +37,49 @@ class ReceiverRunnable(
...
-    val mqReceiver = mq.createReceiver()
+  val timeout = 60.seconds
+  val timeoutNanos = timeout.toNanos
 
-    val start = System.currentTimeMillis()
-    var lastReceived = System.currentTimeMillis()
-    var totalReceived = 0
+  val metricRegistry = new MetricRegistry()
+  val msgTimer = metricRegistry.timer("kafka-receiver-messages-timer")
+  val histogram = metricRegistry.histogram("kafka-receiver-messages-histogram")
```
Comment (mkubala) [inline src/main/scala/com/softwaremill/mqperf/Receiver.scala:47]: Please replace "kafka" with some generic name or use test config to obtain the right prefix.

File: src/main/scala/com/softwaremill/mqperf/Receiver.scala [hunk 2]
```diff
@@ -31,36 +37,49 @@ class ReceiverRunnable(
...
+          lastReceivedNano = nowNano
+          (0 to received).foreach(_ => histogram.update(nowSeconds))
+        }
       }
-
-      totalReceived += received
+      logger.info(s"Test finished, last message read $timeout ago")
+      TestMetrics.receive(metricRegistry).foreach(reportResults.report)
+    }
+    finally {
```
Comment (mkubala) [inline src/main/scala/com/softwaremill/mqperf/Receiver.scala:67]: :+1:

File: src/main/scala/com/softwaremill/mqperf/Receiver.scala [hunk 3]
```diff
@@ -31,36 +37,49 @@ class ReceiverRunnable(
     mq: Mq,
     reportResults: ReportResults,
     receiveMsgBatchSize: Int
-) extends Runnable {
+) extends Runnable with StrictLogging {
 
-  override def run(): Unit = {
-    val mqReceiver = mq.createReceiver()
+  val timeout = 60.seconds
+  val timeoutNanos = timeout.toNanos
```
Comment (mkubala) [inline src/main/scala/com/softwaremill/mqperf/Receiver.scala:43]: ❤️

**Review by mkubala — APPROVED** [review]

File: src/main/scala/com/softwaremill/mqperf/Receiver.scala (hunk 1 above)
Comment (kciesielski) [reply src/main/scala/com/softwaremill/mqperf/Receiver.scala:47]: Of course.

**Pull request merged**